### ISSUE TITLE
exec_publish_done kills its script

### ISSUE THREAD

**Issue by triangle-stream** [arut/nginx-rtmp-module#1508]
Hello,

I'm running nginx 1.17.0 with the rtmp module on Ubuntu 18.04 server.
I'm using the exec_push and exec_publish_done directives in order to run  bash scripts (.sh) i made for.

The exec_publish is working good and launch a bash scritpt (start.sh) that starts an FFmpeg process. It correctly sends the kill signal when publishing stops.

The exec_publish_done, then, launches another bash script (stop.sh) that starts another FFmpeg process but at the same time it starts and immediately stops. 
Manually running stop.sh is currently working good.

My question is: is it possible that the exec_publish_done also sends a SIGTERM or kill immediately after the execution of that script?

I also noticed that this behaviour seems not to be present when i test it after closing and restarting Nginx (stop nginx -> start nginx -> start publish -> exec_publish works -> stop publish -> exec_publish_done works -> restart publish -> exec_publish works -> stop publish -> exec_publish_done does not work anymore).

Same thing also by changinx nginx version (1.14 and 1.15) and ffmpeg version.

**Comment by pyk** [comment]
Yes it is possible. You can use [signal traps](https://github.com/arut/nginx-rtmp-module/wiki/Exec-wrapper-in-bash).
